src/test/unit/algorithms/ConnectionsPerformanceTest.cpp

```diff
@@ -78,22 +83,28 @@ float runTemporalMemoryTest(UInt numColumns, UInt w,   int numSequences,
     for (auto sequence : sequences) {
       for (auto sdr : sequence) {
         tm.compute(sdr, true);
-	//TODO get untrained anomaly score here
+	avgAnomAfter = anom10.compute(tm.anomaly); //average anomaly score
       }
       tm.reset();
     }
+    NTA_CHECK(avgAnomBefore >= avgAnomAfter) << "TM should learn and avg anomalies improve, but we got: "
+      << avgAnomBefore << " and now: " << avgAnomAfter; //invariant
+    avgAnomBefore = avgAnomAfter; //update
   }
   cout << (float)timer.getElapsed() << " in " << label << ": initialize + learn"  << endl;
 
   // test
   for (auto sequence : sequences) {
     for (auto sdr : sequence) {
       tm.compute(sdr, false);
-      //TODO get trained (lower) anomaly
+      avgAnomAfter = anom10.compute(tm.anomaly);
     }
     tm.reset();
   }
-  //TODO check anomaly trained < anomaly untrained
+
+#if defined NDEBUG && !defined(NTA_OS_WINDOWS) //because Win & Debug run shorter training due to time, so learning is not as good
+  NTA_CHECK(avgAnomAfter <= 0.021f) << "Anomaly scores diverged: "<< avgAnomAfter;
+#endif
```

Comment:
Too bad that Windows needs to be excluded but it does run slower than Linux.

Reply:
Thank you for review! 

> Too bad that Windows needs to be excluded but it does run slower than Linux.

yes, on the other hand there's very likely no harm in skipping the test in Win. The test validates the algorithm works, and once it does this will work on all platforms. (esp once we have deterministic checks covered #194 , then we essentially could run tests on just one platform, and then only proper determ. checks)